app/components/documents/document-preview-sidebar.js

```diff
@@ -23,13 +39,88 @@ export default class DocumentsDocumentPreviewSidebar extends Component {
 
   @task
   *loadData() {
-    const docContainer = yield this.args.piece.documentContainer;
-    this.documentType = yield docContainer.type;
+    this.docContainer = yield this.args.piece.documentContainer;
+    this.loadVersionHistory.perform();
+    this.documentType = yield this.docContainer.type;
     this.accessLevel = yield this.args.piece.accessLevel;
   }
 
+  @task
+  *loadVersionHistory() {
+    this.versionPieces = yield this.docContainer.reverseSortedPieces;
+    this.lastPiece = yield this.docContainer.lastPiece;
```

Comment:
Probeer zo veel mogelijk weg te blijven van het gebruik van de async computed properties op de ember-data models. Het verleden heeft al geillustreerd dat dit leidt tot ongecontroleerde data stromen en network requests. `this.docContainer.reverseSortedPieces` is immers heel snel gebruikt zonder de impact in te schatten van de hoeveelheid network requests dat hierdoor mogelijk getriggerd worden. Bovendien zijn heel veel van die computed properties ook weer afhankelijkheden van andere computed properties waardoor ook deze weer getriggerd worden enz.

Je kan best kijken hoe de history opgebouwd wordt in `Documents::DocumentCard` en een gelijkaardige werkwijze toepassen.